pkg/actions/lua/encoding/yaml/yaml.go

```diff
@@ -0,0 +1,52 @@
+package yaml
+
+import (
+	"bytes"
+
+	"github.com/Shopify/go-lua"
+	"github.com/treeverse/lakefs/pkg/actions/lua/util"
+	"gopkg.in/yaml.v3"
+)
+
+func Open(l *lua.State) {
+	yamlOpen := func(l *lua.State) int {
+		lua.NewLibrary(l, yamlLibrary)
+		return 1
+	}
+	lua.Require(l, "encoding/yaml", yamlOpen, false)
+	l.Pop(1)
+}
+
+var yamlLibrary = []lua.RegistryFunction{
+	{Name: "marshal", Function: yamlMarshal},
+	{Name: "unmarshal", Function: yamlUnmarshal},
+}
+
+func check(l *lua.State, err error) {
+	if err != nil {
+		lua.Errorf(l, err.Error())
+		panic("unreachable")
```

Comment:
How expensive is this?  And... are we sure we are exception-safe and `defer x.Close()` for all our Closers along the stack?

Reply:
The lua implementation we are using implemented the error handling using raise (panic), as described https://pkg.go.dev/github.com/Shopify/go-lua#Errorf.

Can go over the current provided files and lookup for resource management issues.